lib/chef/application/client.rb

```diff
@@ -384,6 +338,125 @@ def run_application
   end
 
   private
+
+  def configure_local_mode
+    update_local_mode
+    update_chef_repo_path
+    fetch_local_mode_recipes!
+  end
+
+  def verify_audit_mode!
+    mode = config[:audit_mode] || Chef::Config[:audit_mode]
+
+    return unless mode
+
+    assert_valid_audit_mode!(mode)
+    # This should be removed when audit-mode is enabled by default/no longer
+    # an experimental feature.
+    Chef::Log.warn(audit_mode_experimental_message) unless mode == :disabled
+  end
+
+  def assert_valid_audit_mode!(mode)
+    error_message = unrecognized_audit_mode(mode)
+    valid_mode = %i(enabled disabled audit_only).include?(mode)
+
+    Chef::Application.fatal!(error_message) unless valid_mode
+  end
+
+  def update_chef_client_json
+    json = Chef::Config[:json_attribs]
+
+    return unless json
+
+    @chef_client_json = Chef::ConfigFetcher.new(json).fetch_json
+  end
+
+  def verify_forked_interval!
+    Chef::Application.fatal!(unforked_interval_error_message) if
+      !Chef::Config[:client_fork] && Chef::Config[:interval] &&
+      !Chef::Platform.windows?
+  end
+
+  def update_interval_and_splay
+    interval_key = :interval
+
+    if Chef::Config[:daemonize]
+      Chef::Config[interval_key] ||= 1800
+    elsif Chef::Config[:once]
+      Chef::Config[interval_key] = nil
+      Chef::Config[:splay] = nil
+    end
+  end
+
+  def update_chef_zero
+    host_key = :chef_zero_host
+    port_key = :chef_zero_port
+
+    Chef::Config.chef_zero.host = config[host_key] if config[host_key]
+    Chef::Config.chef_zero.port = config[port_key] if config[port_key]
+  end
+
+  def fetch_local_mode_recipes!
+    return unless Chef::Config.has_key?(:recipe_url)
+
+    exit_status = 1
+
+    Chef::Application.fatal!(
+      'chef-client recipe-url can be used only in local-mode', exit_status
+    ) unless Chef::Config.local_mode
+    extract_recipe_tarball
```

Comment:
And the problem with the inline returns way at the top of the function is that it looks like this function always calls extract_recipe_tarball and i have to scan up to the top to see the return.  if it was enclosed in an `if..end` it would be much easier to see that it was only conditionally executed.